Source/Core/Common/HttpRequest.cpp

```diff
@@ -190,10 +209,24 @@ static size_t CurlWriteCallback(char* data, size_t size, size_t nmemb, void* use
   return actual_size;
 }
 
+static size_t header_callback(char* buffer, size_t size, size_t nitems, void* userdata)
+{
+  auto* headers = static_cast<HttpRequest::Headers*>(userdata);
+  const std::string full_header{buffer, size};
```

Comment:
Could this be a `string_view`?

Reply:
I chose to make it an `std::string` because the `Headers` type is an alias to `std::map<std::string, std::optional<std::string>>`.

I *could* make it an `std::string_view` then convert the resulting substr's to `std::string` to satisfy the `headers->insert` function on line `221`.